<!-- Adapted from franka_ros/franka_control -->
<launch>
  
  <!-- The below machine file makes control PC the 
  "default" machine to execute nodes in. So it is not required to
  explicitly mention machine tag in the nodes that are called below.-->
  <include file="$(find events_shaping_controller_control)/launch/setup.machine" />
  
  <arg name="robot_ip" />
  <arg name="output_type" default="screen" />

  <param name="robot_description" command="$(find xacro)/xacro $(find events_shaping_controller_franka_description)/robots/panda_arm.urdf.xacro" />

  <include file="$(find franka_gripper)/launch/franka_gripper.launch">
    <arg name="robot_ip" value="$(arg robot_ip)" />
  </include>

  <node name="franka_control" pkg="franka_control" type="franka_control_node" output="$(arg output_type)" required="true">
    <rosparam command="load" file="$(find events_shaping_controller_control)/cfg/franka_control_node.yaml" />
    <param name="robot_ip" value="$(arg robot_ip)" />
  </node>

  <!-- NOTE: If franka_control.launch is launched alone, it'll use default parameters for state controllers -->
  <node name="state_controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="$(arg output_type)" args="franka_state_controller"/>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="$(arg output_type)"/>
  <node name="joint_state_publisher" type="joint_state_publisher" pkg="joint_state_publisher" output="$(arg output_type)">
    <rosparam param="source_list">[franka_state_controller/joint_states, franka_gripper/joint_states] </rosparam>
    <param name="rate" value="30"/>
  </node>
  <!-- <node name="joint_state_desired_publisher" type="joint_state_publisher" pkg="joint_state_publisher" output="$(arg output_type)">
    <rosparam param="source_list">[franka_state_controller/joint_states_desired, franka_gripper/joint_states] </rosparam>
    <param name="rate" value="30"/>
    <remap from="/joint_states" to="/joint_states_desired" />
  </node> -->
</launch>
